nixos/modules/hardware/video/nvidia.nix

```diff
@@ -572,11 +572,15 @@ in
           boot = {
             extraModulePackages = if cfg.open then [ nvidia_x11.open ] else [ nvidia_x11.bin ];
             # nvidia-uvm is required by CUDA applications.
-            kernelModules = lib.optionals config.services.xserver.enable [
-              "nvidia"
-              "nvidia_modeset"
-              "nvidia_drm"
-            ];
+            kernelModules = lib.optionals config.services.xserver.enable (
+              [
+                "nvidia"
+                "nvidia_modeset"
+                "nvidia_drm"
+              ]
+              # See https://github.com/NixOS/nixpkgs/issues/334180
+              ++ lib.optionals cfg.open [ "nvidia_uvm" ]
```

Comment:
`cfg.open && config.services.xserver.enable` then, I suppose; a short tldr for the url would be nice, like "softdep observed not to work reliably with the open driver"

Reply:
@SomeoneSerge Do you think this is better?

```nix
kernelModules =
  lib.optionals config.services.xserver.enable [
    "nvidia"
    "nvidia_modeset"
    "nvidia_drm"
  ]
  # With the open driver, nvidia-uvm does not automatically load as
  # a softdep of the nvidia module, so we need to explicitly load it for now.
  # See https://github.com/NixOS/nixpkgs/issues/334180
  ++ lib.optionals (config.services.xserver.enable && cfg.open) [ "nvidia_uvm" ];
```

Reply:
Yes it's ok

Reply:
s/need to /